robocluster/device.py

```diff
@@ -111,6 +114,32 @@ def sleep(self, duration):
             for callback in self.events[event]:
                 self._loop.create_task(callback(event, data))
 
+    async def _serial_read_task(self):
+        """
+        Recieve packets from a linked serial device
+        and call the appropriate callbacks.
+        """
+        while not self._serial_device.isInitialized():
+            await self.sleep(0.01)  # Better way to do this?
+        while True:
+            packet = await self._serial_device.read_packet()
+            event, data = packet['event'], packet['data']
+            for callback in self.events[event]:
+                self._loop.create_task(callback(event, data))
+
+    def link_serial(self, serialdevice):
+        """Integrate an existing serial device I/O for pub/sub communication."""
+        if serialdevice._format not in ['json']:
+            raise RuntimeError('Only json devices are supported at this time')
+        self._serial_device = serialdevice
+        self._serial_device._loop = self._loop
+        # Reinitialize the connection with the new event loop
+        self._loop.create_task(self._serial_device.init_serial())
```

Comment:
This can be called from outside the event loop's thread so it should be using `call_soon_threadsafe` instead of `create_task` as it is not threadsafe.